---
title: "Glotzbach aortopathy RNA-seq"
output: html_notebook
editor_options: 
  chunk_output_type: inline
---


```{r, echo=FALSE}
library(dplyr)
library(tidyr)
library(tidyverse)
```

```{r metadata}
# import the metadata
m1 = readxl::read_xlsx("data/MetadataSamples_QuinlanLab_5.27.xlsx")
m2 = readxl::read_xlsx("data/Glotzbach Sample Match List 05312022.xlsx")

# we care about the GLO ID, ID and the phenotype
m1 = m1[complete.cases(m1),] # drop empty row
m2 = m2 %>% select_if(function(x) !(all(is.na(x))))
m2 = m2 %>% filter(!is.na(Organism))

colnames(m1) = make.names(colnames(m1),unique=TRUE)
colnames(m2) = make.names(colnames(m2),unique=TRUE)

print((m1 %>% left_join(m2, by = c("Glotzbach.Lab.ID" = "Sample.Name"))))

```



```{r deseq2}
library(tximport)
library(DESeq2)

ifiles = list.files("data/rsem/",
                    pattern = "*.isoforms.results")
gfiles = list.files("data/rsem/",
                    pattern = "*.genes.results")

files = paste0("data/rsem/",ifiles)
names(files) = gsub("[:_:].*$","",ifiles)

tx2gene = data.table::fread("data/rsem/18890X1_210521_A00421_0326_BH3HT3DSX2_S1_L001.isoforms.results") %>% select(transcript_id, gene_id)

txi.rsem = tximport(files,type = "rsem", 
                    tx2gene = tx2gene)

sampleTable = data.frame("batch"=gsub("X.*$","",colnames(txi.rsem$counts)))
rownames(sampleTable) = colnames(txi.rsem$counts)

dds <- DESeq2::DESeqDataSetFromTximport(txi.rsem, sampleTable, ~batch)
```

```{r pca}
vsd <- vst(dds, blind=FALSE)
plotPCA(vsd, intgroup = "batch")

```

```{r difexp}
dds = DESeq(dds)
res = results(dds)
res
hist(res$log2FoldChange)
```



```{r}
samp18890 = m1 %>% left_join(m2, by = c("Glotzbach.Lab.ID" = "Sample.Name")) %>%
  select(Phenotype, ID, Glotzbach.Lab.ID) %>%
  filter(grepl("18890",ID)) %>% distinct()


samp18890

```